mpi-proxy-split/p2p_drain_send_recv.cpp

```diff
@@ -257,6 +247,16 @@ drainSendRecv()
     // If any messages are completed aftera  5 seconds sleep,
     // go back to the loop.
   }
+#endif
+  registerLocalSendsAndRecvs();
+  while (global_sent_messages > global_recv_messages) {
+    // If pending MPI_Irecv or MPI_Isend, use MPI_Test to try to complete it.
+    completePendingP2pRequests();
+    // If MPI_Irecv not posted but msg was sent, use MPI_Iprobe to drain msg.
+    drainRemainingP2pMsgs();
+    // Update global recv coutner.
+    registerLocalSendsAndRecvs();
+  }
 }
 
 // FIXME: isBufferedPacket and consumeBufferedPacket both search
```

Comment:
I know that the name `isBufferedPacket()` was here before the PR.  But let's change the name now, to make it more redable.
Maybe we can think of a better name.  A packet is a network packet that is only part of a message.  I propose `MsgBuffer` instead of `BufferedPacket`.
Since we are inspecting messages in g_message_queue, we should probably say `Message` or `Msg` instead of `Packet`.

How about using a name like:
> `existsMatchingMsgBuffer()`
And then we should also change `consumeBufferedPacket` to:
> `consumeMatchingMsgBuffer()`

[ I write consume _Matching_ msg since that function also checks for a match. ]